TOPIC8-MONGODB/src/main/java/data/NotesDao.java

```diff
@@ -0,0 +1,16 @@
+package data;
+
+import java.util.List;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.mongodb.repository.Query;
+
+import models.Notes;
+import models.Student;
+
+public interface NotesDao extends MongoRepository<Notes, String> {
+
+	@Query("{ $and: [ {partialNote1:{$gt: 4}}, {partialNote2:{$gt: 4}},{partialNote3:{$gt: 4}},{finalNote:{$gt: 4}} ]}")

```

Comment:
It looks good, but according the key point 2 (Fetch all students whose notes in a specific course were greater than 4) shouldn't be only comparing the finalNote?